deploy/openshift4/nsx-ipi-helper.yaml

```diff
@@ -0,0 +1,49 @@
+# nsx-ipi-helper DaemonSet
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: nsx-ipi-helper
+  namespace: nsx-system-operator
+  labels:
+    tier: nsx-networking
+    component: nsx-ipi-helper
+    version: v1
+spec:
+  selector:
+    matchLabels:
+      tier: nsx-networking
+      component: nsx-ipi-helper
+      version: v1
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        tier: nsx-networking
+        component: nsx-ipi-helper
+        version: v1
+    spec:
+      hostNetwork: true
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+        - key: node.kubernetes.io/not-ready
+          effect: NoSchedule
+        - key: node.kubernetes.io/unreachable
+          effect: NoSchedule
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      initContainers:
+        - name: nsx-ipi-helper
+          image: registry.access.redhat.com/ubi8/ubi:latest
+          imagePullPolicy: IfNotPresent
+          command: ["/bin/bash", "-c", "yum -y install iptables; echo Disabling the DNS in localhost; iptables -A INPUT -p udp --dport 53 -j DROP; iptables -S INPUT; while ! curl localhost:6443 ; do sleep 1 ; done; echo Enabling the DNS in localhost; iptables -D INPUT -p udp --dport 53 -j DROP; iptables -S INPUT; echo Done"]
```

Comment:
Is only UDP 53 need to be deactivated/activated? What about TCP? Also, not sure if port 9153 should also be added here (likely not)

Reply:
In my verification only deactivating port 53 could work (9153 is not in listening in a running cluster, not sure if it was up during bootstrap), e.g.

> [root@cluster46-q5ntm-worker-hs5sc ~]# **dig api.cluster46.openshift.test**
> 
> ; <<>> DiG 9.11.13-RedHat-9.11.13-6.el8_2.1 <<>> api.cluster46.openshift.test
> ;; global options: +cmd
> ;; Got answer:
> ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 11746
> ;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1
> ;; WARNING: recursion requested but not available
> 
> ;; OPT PSEUDOSECTION:
> ; EDNS: version: 0, flags:; udp: 4096
> ; COOKIE: ab76f2aec2f0ad74 (echoed)
> ;; QUESTION SECTION:
> ;api.cluster46.openshift.test.  IN      A
> 
> ;; ANSWER SECTION:
> api.cluster46.openshift.test. 14 IN     A       192.169.41.11
> 
> ;; Query time: 11 msec
> ;; SERVER: 192.169.41.101#53(192.169.41.101)
> ;; WHEN: Thu Mar 04 06:47:11 UTC 2021
> ;; MSG SIZE  rcvd: 113
> 
> [root@cluster46-q5ntm-worker-hs5sc ~]# **iptables -A INPUT -p udp --dport 53 -j DROP**
> [root@cluster46-q5ntm-worker-hs5sc ~]# **dig api.cluster46.openshift.test**                                                                                                                           
> 
> ; <<>> DiG 9.11.13-RedHat-9.11.13-6.el8_2.1 <<>> api.cluster46.openshift.test
> ;; global options: +cmd
> ;; Got answer:
> ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 6058
> ;; flags: qr aa rd ra; QUERY: 1, ANSWER: 10, AUTHORITY: 2, ADDITIONAL: 4
> 
> ;; OPT PSEUDOSECTION:
> ; EDNS: version: 0, flags:; udp: 4096
> ; COOKIE: 61224b19821debf5901fffa9604082955c325f251bd61990 (good)
> ;; QUESTION SECTION:
> ;api.cluster46.openshift.test.  IN      A
> 
> ;; ANSWER SECTION:
> api.cluster46.openshift.test. 604800 IN A       192.169.41.105
> api.cluster46.openshift.test. 604800 IN A       192.169.41.108
> api.cluster46.openshift.test. 604800 IN A       192.169.41.103
> api.cluster46.openshift.test. 604800 IN A       192.169.41.106
> api.cluster46.openshift.test. 604800 IN A       192.169.41.101
> api.cluster46.openshift.test. 604800 IN A       192.169.41.104
> api.cluster46.openshift.test. 604800 IN A       192.169.41.102
> api.cluster46.openshift.test. 604800 IN A       192.169.41.107
> api.cluster46.openshift.test. 604800 IN A       192.169.41.109
> api.cluster46.openshift.test. 604800 IN A       192.169.41.110
> 
> ;; AUTHORITY SECTION:
> openshift.test.         604800  IN      NS      dns1.openshift.test.
> openshift.test.         604800  IN      NS      localhost.
> 
> ;; ADDITIONAL SECTION:
> dns1.openshift.test.    604800  IN      A       10.92.204.129
> localhost.              604800  IN      A       127.0.0.1
> localhost.              604800  IN      AAAA    ::1
> 
> ;; Query time: 1 msec
> ;; SERVER: 10.40.121.149#53(10.40.121.149)
> ;; WHEN: Thu Mar 04 06:47:46 UTC 2021
> ;; MSG SIZE  rcvd: 347